src/endpoint/s3/s3_bucket_policy_utils.js

```diff
@@ -189,7 +201,23 @@ function _is_principal_fit(account, statement) {
             break;
         }
     }
-    return statement.Principal ? principal_fit : !principal_fit;
+
+    if (statement.NotPrincipal) {
```

Comment:
So after this change, the previous code is only relevant for `statement.Principal`?
If yes, then why did you keep this `let statement_principal = statement.Principal || statement.NotPrincipal;`?

Reply:
If you see `statement_principal` is has in general considered to iterate through it irrespective of `Principal` or `NotPrincipal`. It is not required to be changes I feel